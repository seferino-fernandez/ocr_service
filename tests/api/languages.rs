use axum::{body::Body, http::Request, http::StatusCode};
use http_body_util::BodyExt as _;

use crate::helpers::*;

#[tokio::test]
async fn test_languages_ok() {
    let app = TestApp::new();

    let req = Request::get("/api/v1/languages")
        .body(Body::empty())
        .unwrap();
    let response = app.request(req).await;
    let headers = response.headers().clone();

    assert_eq!(response.status(), StatusCode::OK);
    assert_eq!(headers.get("access-control-allow-origin").unwrap(), "*");
    assert!(headers.get("vary").is_some());
    let body = response.into_body().collect().await.unwrap().to_bytes();
    let body: serde_json::Value = serde_json::from_slice(&body).unwrap();
    insta::assert_yaml_snapshot!(body);
}
