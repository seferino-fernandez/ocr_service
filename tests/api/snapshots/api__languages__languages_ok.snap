---
source: tests/api/languages.rs
expression: body
---
languages:
  - full_path: tesseract/chi_sim/chi_sim.traineddata
    language: chi_sim
    model: chi_sim
    relative_path: chi_sim/chi_sim
  - full_path: tesseract/chi_sim/chi_sim_vert.traineddata
    language: chi_sim
    model: chi_sim_vert
    relative_path: chi_sim/chi_sim_vert
  - full_path: tesseract/eng.traineddata
    language: eng
    model: ~
    relative_path: eng
